spec/helper_spec.rb

```diff
@@ -56,8 +68,22 @@ def task(name: nil)
 describe 'EchoTask' do
   it 'runs an echo task' do
     allow(STDIN).to receive(:read).and_return('{"name": "Lucy"}')
-    expect(EchoTask).to receive(:run).and_return('{"result":
-                                                 "Hello, my name is Lucy"}')
+    out = JSON.dump('result' => 'Hi, my name is Lucy')
+    expect(STDOUT).to receive(:print).with(out)
     EchoTask.run
   end
 end
+
+describe 'SymbolizeTask' do
+  it 'runs an echo task' do
+    params = {
+      'top_level' => { 'nested_key' => 'foo' },
+      'array_keys' => [{ 'array_key' => 'bar' }]
+    }
+    symbols = { nested_hash: 'foo', array_hash: 'bar' }
+    allow(STDIN).to receive(:read).and_return(JSON.dump(params))
+    out = JSON.dump('result' => JSON.dump(params.merge(symbols)))
```

Comment:
I'm not sure why this returns the merged hashes, instead of just the keywords? This works for me locally as expected, but I don't quite follow what this test is doing:
```ruby
lucy@ophelia:~/githubs/bolt/spec (master)$ irb
2.4.4 :001 >     params = {
2.4.4 :002 >           'top_level' => { 'nested_key' => 'foo' },
2.4.4 :003 >           'array_keys' => [{ 'array_key' => 'bar' }]
2.4.4 :004?>       }
 => {"top_level"=>{"nested_key"=>"foo"}, "array_keys"=>[{"array_key"=>"bar"}]} 
2.4.4 :005 > symbols = { nested_hash: 'foo', array_hash: 'bar' }
 => {:nested_hash=>"foo", :array_hash=>"bar"} 
2.4.4 :006 > params.merge(symbols)
 => {"top_level"=>{"nested_key"=>"foo"}, "array_keys"=>[{"array_key"=>"bar"}], :nested_hash=>"foo", :array_hash=>"bar"} 
2.4.4 :007 >
```

Reply:
I updated the code with some comments describing the test. Basically I wanted to test two things 1. symbolizing keys does not destroy structure of input hash 2. the hash in the task has all relevant keys symbolized. So the idea was to 1. return the data hash that was passed in and 2. show that keys are accessed as symbols as expected.